src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -1772,6 +1799,101 @@ class ObservationStore(
     }
   }
 
+  fun recalculateSurvivalRates(monitoringPlotId: MonitoringPlotId) {
+    if (
+        with(MONITORING_PLOTS) {
+          dslContext.fetchExists(
+              this,
+              ID.eq(monitoringPlotId).and(PERMANENT_INDEX.isNull),
+          )
+        }
```

Comment:
It'd be slightly more robust to invert this like `if (!dslContext.fetchExists(this, ID.eq(monitoringPlotId).and(PERMANENT_INDEX.isNotNull)))` so it also fails if the plot doesn't exist at all.

And an optional nit: moving this to a little helper method would make the `if` a bit easier to read.